hack/compare-k8s-feature-gates.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+#
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+usage() {
+  echo "Usage:"
+  echo "> compare-k8s-feature-gates.sh [ -h | <old version> <new version> ]"
+  echo
+  echo ">> For example: compare-k8s-feature-gates.sh 1.26 1.27"
+
+  exit 0
+}
+
+if [ "$1" == "-h" ] || [ "$#" -ne 2 ]; then
+  usage
+fi
+
+versions=("$1" "$2")
+out_dir=$(mktemp -d)
+function cleanup_output {
+    rm -rf "$out_dir"
+}
+trap cleanup_output EXIT
+
+for version in "${versions[@]}"; do
+  versions_dir=test/compatibility_lifecycle/reference
+  
+  # TODO Drop this when Kubernetes v1.34 gets released
+  if [ "$version" == "1.32" ]; then
+    versions_dir="test/featuregates_linter/test_data"
+  fi
+
+ # Get the list of versioned and unversioned feature gates. After v1.34+ unversioned_feature_list.yaml should most likely be dropped
+  wget -q -O - "https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/${versions_dir}/versioned_feature_list.yaml" > "${out_dir}/versioned_featuregates_${version}.yaml"
+  wget -q -O - "https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/${versions_dir}/unversioned_feature_list.yaml" > "${out_dir}/unversioned_featuregates_${version}.yaml"
```

Comment:
What is the purpose of unversioned feature gates? After https://github.com/kubernetes/kubernetes/commit/088daf472b7d1ed4502aef211efca6653b05ff59 it is obsolete and no longer generated/?

Does it contain feature gates which are not bound to any K8s version? I am confused about this file.

Reply:
Yes and v1.32 contains unversioned feature gates
```
❯ cat debug/unversioned_featuregates_1.32.yaml 
- name: ContextualLogging
  versionedSpecs:
  - default: true
    lockToDefault: false
    preRelease: Beta
    version: ""
- name: LoggingAlphaOptions
  versionedSpecs:
  - default: false
    lockToDefault: false
    preRelease: Alpha
    version: ""
- name: LoggingBetaOptions
  versionedSpecs:
  - default: true
    lockToDefault: false
    preRelease: Beta
    version: ""
```

Reply:
If we expect for the script to be run for the next added support v1.34 and onwards what do you think about dropping `unversioned_featuregates`?